code/game/objects/items/stacks/golem_food/golem_status_effects.dm

```diff
@@ -132,12 +159,49 @@
 	ADD_TRAIT(owner, TRAIT_NOHUNGER, TRAIT_STATUS_EFFECT(id))
 	owner.remove_movespeed_modifier(/datum/movespeed_modifier/golem_hunger)
 	owner.remove_status_effect(/datum/status_effect/golem_statued) // Instant fix!
+	set_arm_effects()
 	return TRUE
 
 /datum/status_effect/golem/uranium/on_remove()
 	REMOVE_TRAIT(owner, TRAIT_NOHUNGER, TRAIT_STATUS_EFFECT(id))
+	reset_arm_effects()
 	return ..()
 
+/datum/status_effect/golem/uranium/set_arm_effects()
+	. = ..()
+	RegisterSignals(owner, list(COMSIG_CARBON_BODYPART_ATTACK_PERFORMED, COMSIG_CARBON_BODYPART_CONTACT_PERFORMED), PROC_REF(on_doing_touch))
+
+
+/datum/status_effect/golem/uranium/reset_arm_effects()
+	UnregisterSignal(owner, list(COMSIG_CARBON_BODYPART_ATTACK_PERFORMED, COMSIG_CARBON_BODYPART_CONTACT_PERFORMED))
+	. = ..()
+
+/*
+* on_doing_touch, called by COMSIG_CARBON_BODYPART_ATTACK_PERFORMED or COMSIG_CARBON_BODYPART_CONTACT_PERFORMED
+* Sourced from the most untamed wildernesses of the unarmed attack chain
+* This is only called if someone doesn't successfully block, so shields will protect you
+* but armor won't (however, rad-safe clothing will)
+*/
+/datum/status_effect/golem/uranium/proc/on_doing_touch(datum/source, mob/living/carbon/human/touchy_golem, mob/living/target, obj/item/bodypart/active_bodypart, damage_dealt = 0, obj/item/bodypart/targeted_bodypart = null)
+	SIGNAL_HANDLER
+	if(!CAN_IRRADIATE(target))
+		return
+	if(!modified_arms.Find(active_bodypart)) // maybe you got your arms replaced by a bioscrambler in the time since you got this effect
+		return
+	if(!touchy_golem.combat_mode)
+		if(prob(80)) // 20% chance to irradiate people anyway even if you're only trying to help
+			return	// why does your love kill?
+	if(SSradiation.irradiate(target))
+		target.visible_message(
+		span_danger("Growing green flecks fall from [owner]'s hands onto [target]."),
+		span_priorityalert("After [owner] touches you, your skin starts burning and glowing!"),
+		vision_distance = COMBAT_MESSAGE_RANGE
+		)
+		ASYNC // async so we can scream without upsetting should_not_sleep
+			target.emote("scream")
+	remove_duration(30 SECONDS) // lessen the duration to avoid uranium golems becoming walking chernobyl
```

Comment:
Might be worth warning the golem in some way that this is happening. 
Text in the chat is sufficient, but pulsing the border of the status effect (or some other visual) too would be cool.